### PR TITLE
Changed to use PG eeprom code

### PR DESCRIPTION
**Pull request by martinbudden**

### CODE REVIEW

**Comment by martinbudden** [comment]
@digitalentity ,

the conversion to parameter groups is finally happening.

I've set it up so masterConfig and parameter groups can exist side be side. Well, everywhere except the CLI.

So what I'd like to do is merge each one in as I do it. This has a number of advantages:
1. No need to keep rebasing a massive parameter group PR. This is a pain and is also error-prone.
2. Code can be tested as soon as it is available, so any errors can be tracked down more quickly.

There is a minor downside. As I add PGs, they will stop being available in the CLI (that is until I switch the CLI over to using the new PGs). This means that we will have a period of about a couple of weeks when we don't have everything available in the CLI. (Configurator will work fine though).

What do you think?

File: src/main/config/config_streamer.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    if (c->err != 0) {
+        return c->err;
+    }
+
+    HAL_StatusTypeDef status;
+    if (c->address % FLASH_PAGE_SIZE == 0) {
+        FLASH_EraseInitTypeDef EraseInitStruct = {0};
+        EraseInitStruct.TypeErase     = FLASH_TYPEERASE_SECTORS;
+        EraseInitStruct.VoltageRange  = FLASH_VOLTAGE_RANGE_3; // 2.7-3.6V
+        EraseInitStruct.Sector        = (FLASH_SECTOR_TOTAL-1);
```
Comment (digitalentity) [inline src/main/config/config_streamer.c]: Please change this to respect current approach that is not using macros and relies on linker script parameters.
See `getFLASHSectorForEEPROM()` implementation

File: src/main/config/config_streamer.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+{
+    if (c->err != 0) {
+        return c->err;
+    }
+
+    FLASH_Status status;
+
+    if (c->address % FLASH_PAGE_SIZE == 0) {
+#if defined(STM32F40_41xxx)
+        status = FLASH_EraseSector(FLASH_Sector_8, VoltageRange_3); //0x08080000 to 0x080A0000
```
Comment (digitalentity) [inline src/main/config/config_streamer.c]: Please change this to respect current approach that is not using macros and relies on linker script parameters.
See `getFLASHSectorForEEPROM()` implementation

**Review by digitalentity — CHANGES_REQUESTED** [review]

File: src/main/config/config_streamer.c (hunk 1 above)
Comment (martinbudden) [reply src/main/config/config_streamer.c]: OK

**Comment by martinbudden** [comment]
@digitalentity , can you approve PR and then I'll merge in once I've re-tested.

**Review by digitalentity — APPROVED** [review]

**Pull request merged**